### PR TITLE
test: add test to clients endpoints

### PR DESCRIPTION
**Pull request by fjsandov**

### CODE REVIEW

File: src/routes/clients.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  constants: { CLIENT_KINDS },
+} = require('@notify-watcher/core');
+const request = require('../test/supertest');
+const User = require('../models/user');
+
+describe('clients routes', () => {
+  describe('POST clients/register', () => {
+    const registerUrl = '/clients/register';
+
+    test('for a non existent user', async () => {
```
Comment (negebauer) [inline src/routes/clients.test.js]: `for a non existent user` should be inside a `describe` and then `it('returns 404 error')`

File: src/routes/clients.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      let user;
+
+      beforeAll(async () => {
+        user = await User.create({ email: userEmail });
+        await user.createSecret();
+      });
+
+      afterAll(() => user.deleteOne());
+
+      test('with a invalid token', async () => {
```
Comment (negebauer) [inline src/routes/clients.test.js]:
```js
describe('with an invalid token', () => {
  it('returns a 401 error')
})
```

File: src/routes/clients.test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          .post(registerUrl)
+          .send({
+            email: userEmail,
+            token: 'invalid',
+            clientData,
+          })
+          .expect(401);
+      });
+
+      test('with an invalid client data', async () => {
```
Comment (negebauer) [inline src/routes/clients.test.js]: same as ^

File: src/routes/clients.test.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            email: userEmail,
+            token,
+            clientData: {
+              kind: 'invalid-kind',
+            },
+          })
+          .expect(500);
+      });
+
+      test('with a valid token and client data', async () => {
```
Comment (negebauer) [inline src/routes/clients.test.js]: same as ^

**Comment by negebauer** [comment]
Remember to add PR to the project in the sidebar ->

File: src/routes/clients.test.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        await request
+          .post(registerUrl)
+          .send({
+            email: userEmail,
+            token,
+            clientData: {
+              kind: 'invalid-kind',
+            },
+          })
+          .expect(500);
```
Comment (sasalatart) [inline src/routes/clients.test.js]: Should this really be 400, or some other 4xx?

Comment (negebauer) [reply src/routes/clients.test.js]: yeah, missed it.
Invalid data in the request -> whoever made the request is the problem, not the server

It should be a 4xx

File: src/mongoose.js [hunk 6]
```diff
@@ -3,13 +3,25 @@ const mongoose = require('mongoose');
 // To fix https://github.com/Automattic/mongoose/issues/4291
 mongoose.Promise = global.Promise;
 
-function dbConnect(databaseUrl) {
+/**
+ * Connect to the database.
+ */
+const connect = async databaseUrl => {
```
Comment (sasalatart) [inline src/mongoose.js]: No need for async if there is no `await` inside (you are already returning a promise)

File: src/routes/clients.test.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        user = await User.create({ email: 'user-client-register@example.org' });
+        await user.createSecret();
+        token = await user.generateToken();
+      });
+
+      afterAll(() => user.deleteOne());
+
+      describe('with a invalid token', () => {
+        it('should return "unauthorized"', () =>
+          register({
```
Comment (sasalatart) [inline src/routes/clients.test.js]: If this is async, you should probably `return` this, or not?

Comment (fjsandov) [reply src/routes/clients.test.js]: it is returning it, `it('sho...', () => register().expect(...));`

File: src/routes/clients.test.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
+const {
+  constants: { CLIENT_KINDS },
```
Comment (negebauer) [inline src/routes/clients.test.js]: removed from `core`, now it's in `src/notifications/clients`

File: src/routes/clients.test.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+      let user;
+      let token;
+
+      beforeAll(async () => {
+        user = await User.create({ email: 'user-client-register@example.org' });
+        await user.createSecret();
+        token = await user.generateToken();
+      });
+
+      afterAll(() => user.deleteOne());
```
Comment (negebauer) [inline src/routes/clients.test.js]: we should use `deleteOne` with Model, not instances. So we don't have concussion between `User.deleteOne` and `user.deleteOne`

```suggestion
      afterAll(() => user.remove());
```

File: src/routes/clients.test.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+      describe('with a invalid token', () => {
```
Comment (negebauer) [inline src/routes/clients.test.js]:
```suggestion
      describe('with an invalid token', () => {
```

File: src/routes/clients.test.js (hunk 8 above)
Comment (fjsandov) [reply src/routes/clients.test.js]: Will be resolved in another PR.

**Pull request merged**